snet_cli/utils.py

```diff
@@ -83,3 +85,40 @@ def type_converter(t):
         return web3.Web3.toChecksumAddress
     else:
         return str
+
+
+def _validate_path(path, entry_path):
+    return entry_path.parent in path.parents
+
+
+def _add_next_paths(path, entry_path_parents, walked_paths, next_paths):
+    with open(path) as f:
+        for line in f:
+            if line.strip().startswith("import"):
+                import_statement = "".join(line.split('"')[1::2])
+                if not import_statement.startswith("google/protobuf"):
+                    import_statement_path = PurePath(Path(path.parent.joinpath(import_statement)).resolve())
+                    if _validate_path(import_statement_path, entry_path_parents):
+                        if import_statement not in walked_paths:
+                            next_paths.append(import_statement_path)
+                    else:
+                        raise ValueError("Path must not be a parent of entry path")
+    walked_paths.add(path)
```

Comment:
this still allows for the situation described before

A is `entry_path`
A imports B
A imports C
B imports C

```
walked_paths = []
next_paths = [entry_path] == [A]
path = next_paths.pop() == A => next_paths == []
add_next_paths(A) => next_paths == [B,C], walked_paths == [A]
path = next_paths.pop() == B => next_paths == [C]
add_next_paths(B) => next_paths == [C,C] ***this shouldn't happen***, walked_paths == [A,B]
```